### PR TITLE
Fixes #6829: Unexpected report for SSH process when service is restarted

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/6829

### CODE REVIEW

**Comment by peckpeck** [comment]
Could you check the bug in previous technique versions too?
Thank you

**Comment by amousset** [comment]
The problem does not exist in the previous versions.

File: techniques/systemSettings/remoteAccess/sshConfiguration/4.0/main.st [hunk 1]
```diff
@@ -144,7 +144,7 @@ bundle agent rudder_openssh_server_configuration(class_prefix, service_name, par
 
       "any"
         usebundle  => rudder_common_report("${service_name}", "result_success", "${${params}[report]}", "SSH process", "None", "The ${service_name} service is running"),
-        ifvarclass => "!${class_prefix}_down";
+        ifvarclass => "!${class_prefix}_down.!(service_restart_ssh_reached|service_restart_sshd_reached)";
```
Comment (peckpeck) [inline techniques/systemSettings/remoteAccess/sshConfiguration/4.0/main.st]: isn't there the same bug below, line 213

**Comment by peckpeck** [comment]
But I see the same code present in rudder 2.11 and in sshConfguration version 3

File: techniques/systemSettings/remoteAccess/sshConfiguration/4.0/main.st (hunk 1 above)
Comment (ncharles) [inline techniques/systemSettings/remoteAccess/sshConfiguration/4.0/main.st]: why reached ?
shouldn't it be:
!${class_prefix}_down.!(service_restart_ssh_ok|service_restart_sshd_ok) ?

Comment (amousset) [inline techniques/systemSettings/remoteAccess/sshConfiguration/4.0/main.st]: But is the restart fails, wouldn't this report a success ?

Comment (ncharles) [inline techniques/systemSettings/remoteAccess/sshConfiguration/4.0/main.st]: Ohhhhhhhhhh this is clever !!!

**Pull request merged**